From 3b0bdecfb147ba7c18673fe83b68f25e79fd9ed7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nicol=C3=A1s=20Alvarez?= <nicolas.alvarez@gmail.com>
Date: Mon, 15 Sep 2014 00:22:58 -0300
Subject: [PATCH] Fix build on all supported MSVC versions.

std::function<...> x = 0 doesn't compile on MSVC2012 and 2013,
std::function<...> x = {} doesn't compile on MSVC2010 and 2012,
and std::function<...> x = nullptr apparently won't work in gcc 4.5.
Fix by using std::function<...> x = std::function<...>().
---
 src/lib/plugin/kpluginloader.h | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/lib/plugin/kpluginloader.h b/src/lib/plugin/kpluginloader.h
index ebfb634..855786b 100644
--- a/src/lib/plugin/kpluginloader.h
+++ b/src/lib/plugin/kpluginloader.h
@@ -307,7 +307,8 @@ public:
      * @since 5.1
      */
     static QList<QObject *> instantiatePlugins(const QString &directory,
-            std::function<bool(const KPluginMetaData &)> filter = Q_NULLPTR, QObject* parent = Q_NULLPTR);
+            std::function<bool(const KPluginMetaData &)> filter = std::function<bool(const KPluginMetaData &)>(),
+            QObject* parent = Q_NULLPTR);
 
     /**
      * Find all plugins inside @p directory. Only plugins which have JSON metadata will be considered.
@@ -326,7 +327,7 @@ public:
      * @since 5.1
      */
     static QVector<KPluginMetaData> findPlugins(const QString &directory,
-            std::function<bool(const KPluginMetaData &)> filter = Q_NULLPTR);
+            std::function<bool(const KPluginMetaData &)> filter = std::function<bool(const KPluginMetaData &)>());
 
     /**
      * Invokes @p callback for each valid plugin found inside @p directory. This is useful if
@@ -348,7 +349,7 @@ public:
      * @since 5.1
      */
     static void forEachPlugin(const QString &directory,
-            std::function<void(const QString &)> callback = Q_NULLPTR);
+            std::function<void(const QString &)> callback = std::function<void(const QString &)>());
 private:
     Q_DECLARE_PRIVATE(KPluginLoader)
     Q_DISABLE_COPY(KPluginLoader)
-- 
1.9.0.msysgit.0

